stdlib/bigint.gr

```diff
@@ -0,0 +1,489 @@
+export let lnot = (num: BigInt) => {
+  WasmI32.toGrain(BI.bitwiseNot(WasmI32.fromGrain(num)))
+}
+
+/**
+ * Computes the bitwise AND (`&`) on the given operands.
+ *
+ * @param x: The first operand
+ * @param y: The second operand
+ * @returns Containing a `1` in each bit position for which the corresponding bits of both operands are `1`
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let land = (num1: BigInt, num2: BigInt) => {
+  WasmI32.toGrain(
+    BI.bitwiseAnd(WasmI32.fromGrain(num1), WasmI32.fromGrain(num2))
+  )
+}
+
+/**
+ * Computes the bitwise OR (`|`) on the given operands.
+ *
+ * @param x: The first operand
+ * @param y: The second operand
+ * @returns Containing a `1` in each bit position for which the corresponding bits of either or both operands are `1`
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let lor = (num1: BigInt, num2: BigInt) => {
+  WasmI32.toGrain(
+    BI.bitwiseOr(WasmI32.fromGrain(num1), WasmI32.fromGrain(num2))
+  )
+}
+
+/**
+ * Computes the bitwise XOR (`^`) on the given operands.
+ *
+ * @param x: The first operand
+ * @param y: The second operand
+ * @returns Containing a `1` in each bit position for which the corresponding bits of either but not both operands are `1`
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let lxor = (num1: BigInt, num2: BigInt) => {
+  WasmI32.toGrain(
+    BI.bitwiseXor(WasmI32.fromGrain(num1), WasmI32.fromGrain(num2))
+  )
+}
+
+/**
+ * Counts the number of leading zero bits in the value.
+ * Will return the maximum integer for negative numbers.
+ *
+ * @param value: The value to inspect
+ * @returns The amount of leading zeros
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let clz = (num: BigInt) => {
+  WasmI32.toGrain(
+    DS.newInt32(BI.countLeadingZeros(WasmI32.fromGrain(num)))
+  ): Int32
+}
+
+/**
+ * Counts the number of trailing zero bits in the value.
+ *
+ * @param value: The value to inspect
+ * @returns The amount of trailing zeros
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let ctz = (num: BigInt) => {
+  WasmI32.toGrain(
+    DS.newInt64(BI.countTrailingZeros(WasmI32.fromGrain(num)))
+  ): Int64
+}
+
+/**
+ * Counts the number of bits set to `1` in the value, also known as a population count.
+ * Will return the `None` if given a negative integer
+ *
+ * @param value: The value to inspect
+ * @returns The amount of 1-bits in its operand
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let popcnt = (num: BigInt) => {
+  let flagDest = Memory.malloc(4n)
+  WasmI32.store(flagDest, 0n, 0n)
+  let res = BI.popcnt(WasmI32.fromGrain(num), flagDest)
+  if (WasmI32.eqz(WasmI32.load(flagDest, 0n))) {
+    Some(WasmI32.toGrain(DS.newInt64(res)): Int64)
+  } else {
+    None
+  }
+}
+
+/**
+ * @section Other: Other functions on BigInts.
+ */
+
+/**
+ * Converts the given operand to a string.
+ *
+ * @param num: The operand
+ * @returns The operand, as a string
+ *
+ * @since v0.5.0
+ */
+@unsafe
+export let toString = (num: BigInt) => {
+  BI.bigIntToString10(WasmI32.fromGrain(num))
+}
```

Comment:
I'd say that this is a little different, since it's not converting to/from a `Number` type. Note that the doc comment for the `Conversions` section is
```
Conversions: Functions for converting between Numbers and the BigInt type.
```
(which aligns with the other number types)